actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -134,7 +134,7 @@ module CacheHelper
       #
       #   <%= render @notifications, cache: false %>
       def cache(name = {}, options = nil, &block)
-        if controller.perform_caching
+        if controller.try(:perform_caching)
```

Comment:
We will also not catch it with the new code. Nil#respond_to? is there as
well.

On Tue, Apr 28, 2015 at 9:58 PM, Rafael Mendonça França <
notifications@github.com> wrote:

> In actionview/lib/action_view/helpers/cache_helper.rb
> <https://github.com/rails/rails/pull/19941#discussion_r29281815>:
>
> > @@ -134,7 +134,7 @@ module CacheHelper
> >        #
> >        #   <%= render @notifications, cache: false %>
> >        def cache(name = {}, options = nil, &block)
> > -        if controller.perform_caching
> > +        if controller.try(:perform_caching)
>
> Yeah, but if for some case receiver is not there (and it should always be)
> we will not catch the problem.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/19941/files#r29281815>.
>